File: Challenges/Java/ReverseVowelsOfString.java
public final class ReverseVowelsOfString {
	
	String reverseVowelsOfString(String s) {
	    String output = "", vowels = "";

	    for(char ch : s.toCharArray()) {
	        if((ch+"").matches("[aeiouAEIOU]")) {
	            String ss = ch+vowels;
	            vowels = ss;
	        }
	    }

	    for(int i = 0, x = 0; i < s.length(); i++) {
	        if((s.charAt(i)+"").matches("[aeiouAEIOU]")) {
	            output += vowels.charAt(x);
	            x++;
	            continue;
	        }
	        output += s.charAt(i);
	    }

	    return output;
	}
}

File: Challenges/Javascript/reverseVowelsOfString.js
function reverseVowelsOfString(s) {
    let out = "", vowels = "";

    for(let ch of s) {
        if(ch.match(/[aeiou]/i) == ch) {
            let ss = ch+vowels;
            vowels = ss;
        }
    }

    for(let i = 0, x = 0; i < s.length; i++) {
        let ch = s.charAt(i);
        if(ch.match(/[aeiou]/i) == ch) {
            out += vowels[x];
            x++;
            continue;
        }
        out += ch;
    }
    return out;
}